packages/terser/src/type.ts

```diff
@@ -23,6 +23,12 @@ export interface WorkerOutput {
   sourceMap?: Record<string, any>;
 }
 
+export interface WorkerError {
```

Comment:
this is pretty loose. being more constrained would probably be a good idea. Adding an `error` property to `WorkerOutput` is more semantic. `error` can be an object with `message` and `stack`, and then all you need do is check if `data.error` is truthy on line 111. that will remove some complexity I don't think is needed.

Reply:
@shellscape but that would imply that all other properties of the `WorkerOutput` must be nullable, in case an error occoured and the error property is set and than it would be harder to distinguish between a success and error state.

Reply:
not sure I follow:

`error` is nullish: no error
`error` is truthy: error

Reply:
sure, but the `code` property e.g. could then no longer be a mandatory property.

Reply:
yes. why is that a concern?

Reply:
I like to map different states using different types and then merge them with a union type if necessary :innocent: .
Even with callbacks, there are also two parameters (error, result), instead of the error being an optional property of the result parameter. The same applies to promises. But I have no problem doing it the way you suggested, if you prefer :v: :relaxed: .